src/test/java/org/broadinstitute/hellbender/engine/spark/SparkSharderUnitTest.java

```diff
@@ -0,0 +1,393 @@
+
+import com.google.common.collect.*;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.samtools.util.OverlapDetector;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.PairFlatMapFunction;
+import org.broadinstitute.hellbender.engine.Shard;
+import org.broadinstitute.hellbender.engine.ShardBoundary;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class SparkSharderUnitTest extends BaseTest implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final int STANDARD_READ_LENGTH = 3;
+
+    private SAMSequenceDictionary sequenceDictionary = new SAMSequenceDictionary(
+            ImmutableList.of(new SAMSequenceRecord("1", 100), new SAMSequenceRecord("2", 50)));
+
+
+    @Test
+    public void testLocatablesPerShard() throws IOException {
+
+        // Consider the following reads (divided into four partitions), and intervals.
+        // This test checks that iterating over the reads and intervals at the same time produces the overlapping
+        // reads for each interval.
+
+        //                      1                   2
+        //    1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7
+        // ---------------------------------------------------------
+        // Reads in partition 0
+        //   [-----]
+        //           [-----]
+        //               [-----]
+        // ---------------------------------------------------------
+        // Reads in partition 1
+        //               [-----]
+        //               [-----]
+        //               [-----]
+        // ---------------------------------------------------------
+        // Reads in partition 2
+        //               [-----]
+        //                       [-----]
+        //                         [-----]
+        // ---------------------------------------------------------
+        // Reads in partition 3
+        //                                   [-----]
+        //                                           [-----]
+        //                                                   [-----]
+        // ---------------------------------------------------------
+        // Intervals
+        //     [-----]
+        //                 [---------]
+        //                       [-----------------------]
```

Comment:
What happened to the partition read extents in this diagram (could have sworn they were here when you presented last week)?